sinks/xray/xray.go

```diff
@@ -0,0 +1,220 @@
+package xray
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"math"
+	"net"
+	"regexp"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stripe/veneur/protocol"
+	"github.com/stripe/veneur/sinks"
+	"github.com/stripe/veneur/ssf"
+	"github.com/stripe/veneur/trace"
+	"github.com/stripe/veneur/trace/metrics"
+)
+
+const subsegmentType = "subsegment"
+
+var segmentHeader = []byte(`{"format": "json", "version": 1}` + "\n")
+
+const XRayTagNameClientIP = "xray_client_ip"
+
+type XRaySegmentHTTPRequest struct {
```

Comment:
It looks like these are all created from example input. Can we generate them (and move them into a separate file)?

Reply:
I took a stab at this, but the issue is that the AWS spec for this is rife with JSONy maps of mixed types, mixtures of objects and other stuff. We _could_ move it here, but it feels like more trouble than it's worth to me. [Here's the spec](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html) if you wanna check it out.